bundles/AdminBundle/src/Controller/Admin/DataObject/DataObjectController.php

```diff
@@ -1385,10 +1385,19 @@ public function saveAction(Request $request): JsonResponse
                 $localizedFields = $object->getLocalizedFields();
                 $localizedFields->setLoadedAllLazyData();
             }
+
+            // Mark fields that have changed as dirty
+            if ($request->get('task') !== 'autoSave' && $request->get('task') !== 'unpublish') {
+                $fields = array_keys($object->getClass()->getFieldDefinitions());
+                foreach ($fields as $field) {
+                    $getter  ='get' . ucfirst($field);
+                    if ($object->$getter() !== $objectFromDatabase->$getter()) {
```

Comment:
I am facing a problem where the object and arrays are seen as changed 🤔 the only fix is to convert to array and use `array_diff_assoc`, any ideas?

To reproduce, try to save a new `Customer`, and put a breakpoint in there, it would trigger that `Calculated Segments` has changed.